grouper/fe/handlers/service_account_create.py

```diff
@@ -0,0 +1,73 @@
+from sqlalchemy.exc import IntegrityError
+
+from grouper.fe.forms import ServiceAccountCreateForm
+from grouper.fe.settings import settings
+from grouper.fe.util import GrouperHandler
+from grouper.model_soup import Group
+from grouper.models.audit_log import AuditLog
+from grouper.models.user import User
+
+
+class ServiceAccountCreate(GrouperHandler):
+    def get(self):
+        form = ServiceAccountCreateForm()
+        return self.render(
+            "service-account-create.html", form=form,
+        )
+
+    def post(self):
+        if "@" not in self.request.arguments["name"][0]:
+            self.request.arguments["name"][0] += "@" + settings.service_account_email_domain
+
+        form = ServiceAccountCreateForm(self.request.arguments)
+
+        if not form.validate():
+            return self.render(
+                "service-account-create.html", form=form,
+                alerts=self.get_form_alerts(form.errors)
+            )
+
+        if form.data["name"].split("@")[-1] != settings.service_account_email_domain:
+            form.name.errors.append("All service accounts must have a username ending in {}"
+                .format(settings.service_account_email_domain))
+            return self.render(
+                "service-account-create.html", form=form,
+                alerts=self.get_form_alerts(form.errors)
+            )
+
+        user = User(username=form.data["name"], role_user=True)
```

Comment:
can we push this stuff down into `grouper.service_account`? in additional to separating business logic from display concerns we can reuse it (e.g. ctl or, *gasp*, a write api).